kratos/python_scripts/eigen_solver_factory.py

```diff
@@ -19,23 +19,29 @@ def ConstructSolver(settings):
             return eigen_solver
         else:
             raise Exception("EigenSolversApplication not available")
-    elif solver_type == "eigen_feast" or solver_type == "eigen_feast_complex":
+    elif solver_type == "eigen_feast":
         if kratos_utils.CheckIfApplicationsAvailable("EigenSolversApplication"):
             import KratosMultiphysics.EigenSolversApplication as EiSA
-            if solver_type == "eigen_feast":
-                if settings.Has("symmetric") and settings["symmetric"].GetBool() and hasattr(EiSA, "FEASTSymmetricEigensystemSolver"):
+            try:
```

Comment:
maybe a better solution for the `try except` would be to add a flag to the [`EigenSolversApplication` (the object itself)](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/EigenSolversApplication/eigen_solvers_application.h#L45) to check whether feast is compiled or not => similar to how the [build-type is accessed in the kernel](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/kernel.h#L134)
This would make those checks clear and also help with other things such as the testing

=> my main concern is that the `try except` can hide more errors than it should

@oberbichler what do you think?

Reply:
Maybe like this: 

```cpp
constexpr bool HasMkl()
{
#if defined(USE_EIGEN_MKL)
    return true;
#else
    return false;
#endif
}
    
constexpr bool HasFeast()
{
#if defined(USE_EIGEN_FEAST)
    return true;
#else
    return false;
#endif
}
```

and register them to python?

Reply:
yes that would do the job I suppose

Reply:
I will add it. @oberbichler you can adapt it for MKL then if you need it

Reply:
What should I adapt for MKL?

Reply:
I meant you can use the function for your MKL tests, if you like